src/OMSimulatorLib/Connection.cpp

```diff
@@ -32,10 +32,115 @@
 #include "Connection.h"
 #include "ssd/ConnectionGeometry.h"
 #include "Logging.h"
+#include "ssd/Tags.h"
 
 #include <cstring>
 #include <iostream>
 
+
+oms3::Connection::Connection(const oms3::ComRef& conA, const oms3::ComRef& conB)
+{
+  std::string str;
+
+  this->type = oms3_connection_single;
+
+  str = std::string(conA);
+  this->conA = new char[str.size()+1];
+  strcpy(this->conA, str.c_str());
+
+  str = std::string(conB);
+  this->conB = new char[str.size()+1];
+  strcpy(this->conB, str.c_str());
+
+  this->geometry = reinterpret_cast<ssd_connection_geometry_t*>(new oms2::ssd::ConnectionGeometry());
+}
+
+oms3::Connection::~Connection()
+{
+  if (this->conA) delete[] this->conA;
+  if (this->conB) delete[] this->conB;
+  if (this->geometry) delete reinterpret_cast<oms2::ssd::ConnectionGeometry*>(this->geometry);
+}
+
+oms3::Connection::Connection(const oms3::Connection& rhs)
+{
+  this->type = rhs.type;
+
+  this->conA = new char[strlen(rhs.conA)+1];
+  strcpy(this->conA, rhs.conA);
+
+  this->conB = new char[strlen(rhs.conB)+1];
+  strcpy(this->conB, rhs.conB);
+
+  oms2::ssd::ConnectionGeometry* geometry_ = new oms2::ssd::ConnectionGeometry();
+  *geometry_ = *reinterpret_cast<oms2::ssd::ConnectionGeometry*>(rhs.geometry);
+  this->geometry = reinterpret_cast<ssd_connection_geometry_t*>(geometry_);
+}
+
+oms3::Connection& oms3::Connection::operator=(const oms3::Connection& rhs)
+{
+  // check for self-assignment
+  if(&rhs == this)
+    return *this;
+
+  if (this->type != rhs.type)
+    logWarning("[oms2::Connection::operator=] changing type of connection");
```

Comment:
@adeas31 I guess you are right and we should also populate the function name for warnings automatically to prevent such mistakes.